### DISCUSSION TITLE
Detect and warn the user when using Windows.System

### DISCUSSION THREAD

**Discussion by ArchieCoder** [unoplatform/uno#17203] [General]
I copied-pasted my PortableDispatcher class from my UWP project to Uno.

I was wondering why dispatcherQueue = DispatcherQueue.GetForCurrentThread(); was return null.

It turned out that my using declaration was Windows.System, but it needs to be Microsoft.UI.Dispatching.

Can this be detected at compilation time?

**Comment by jeromelaban** [comment]
It probably could be detected, yes, though on Windows, it will crash on use, suggesting to use the other API. Uno supports it because of its UWP compatibility.
